packages/components/src/palette-edit/index.tsx

```diff
@@ -154,17 +167,31 @@ function Option< T extends Color | Gradient >( {
 	onStartEditing,
 	onRemove,
 	onStopEditing,
+	popoverProps: receivedPopoverProps,
 	slugPrefix,
 	isGradient,
 }: OptionProps< T > ) {
 	const focusOutsideProps = useFocusOutside( onStopEditing );
 	const value = isGradient ? element.gradient : element.color;
 
+	// Use internal state instead of a ref to make sure that the component
+	// re-renders when the popover's anchor updates.
+	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
+	const popoverProps = useMemo(
+		() => ( {
+			...receivedPopoverProps,
+			// Use the custom palette color item as the popover anchor.
+			anchor: popoverAnchor,
+		} ),
+		[ popoverAnchor, receivedPopoverProps ]
+	);
```

Comment:
Is it worth the memoizing overhead when we don't even pass a stable props object in our own app? (in `color-palette-panel.js`)

Reply:
I'm not sure if this is a bit of premature optimization as well. Looking through the rest of the codebase it appears that the majority of use cases memoize the popover props.

I've updated `color-palette-panel.js` to avoid recreating the popover props object and made the ColorPickerPopover memoize its popover props as well for consistency. I can of course remove the useMemo instances if you feel this was the wrong direction.

What do you think?

Reply:
Looks good, let's go with this then 👍